### ISSUE TITLE
Class declaration incorrectly identified as Constructor

### ISSUE THREAD

**Issue by chochos** [ceylon/ceylon-spec#1195]
In this code:

```ceylon
class Foo {
  shared new Foo() {}
  shared new Bar() {}
  shared Foo fail() {
    return Foo.Bar();
  }
}
```

The expression `Foo.Bar` in the method `fail` is a QualifiedTypeExpression that refers to the constructor `Foo.Bar`, but its BaseTypeExpression refers to the constructor `Foo.Foo` when it should refer to the `Foo` class.

This doesn't seem to cause problems in the JVM backend, I suppose they worked around this, but I just found this out when fixing a related bug in JS.

**Comment by gavinking** [comment]
Well this is expected. Is it causing any bugs?

**Comment by chochos** [comment]
I worked around it as well, but files this anyway in case it wasn't expected and was modified.

**Comment by chochos** [comment]
Why is it expected BTW? The `Foo` in `Foo.Bar` should be the class, no the default constructor...

**Comment by gavinking** [comment]
The typechecker doesn't know about the `.Bar` when it analyses `Foo`.

**Comment by gavinking** [comment]
I suppose I could add an error to prevent you from writing the above, forcing you to write plain `Bar()` instead.

**Comment by gavinking** [comment]
Was fixed by the new approach in #1250.
